src/sonic-yang-models/yang-models/sonic-stp.yang

```diff
@@ -0,0 +1,236 @@
+module sonic-stp {
+    namespace "http://github.com/sonic-net/sonic-stp";
+    prefix stp;
+
+    description "SONiC STP YANG model for PVST and MST configurations";
+
+    revision 2024-11-22 {
+        description "Version01 for combined PVST and MST configurations.";
+    }
+
+    container stp {
+        description "STP global and per-instance configurations";
+
+        container global {
+            description "Global STP settings";
+            leaf mode {
+                type enumeration {
+                    enum "pvst" {
+                        description "Per VLAN Spanning Tree mode";
+                    }
+                    enum "mst" {
+                        description "Multiple Spanning Tree mode";
+                    }
+                }
+                description "STP mode (PVST or MST)";
+            }
+            leaf rootguard_timeout {
+                type uint16 {
+                    range "5..600";
+                }
+                default "30";
+                description "Root guard timeout in seconds";
+            }
+            leaf forward_delay {
```

Comment:
grouping vlanModeAttr {
        leaf forward_delay {
            type uint8 {
                range "4..30" {
                    error-message "Invalid Forwarding Delay value.";
                }
            }
            units seconds;
            default 15;
            description
                "The delay used by STP bridges to transition root and
                designated ports to forwarding";
        }

        leaf hello_time {
            type uint8 {
                range "1..10" {
                    error-message "Invalid Hello Time value.";
                }
            }
            units seconds;
            default 2;
            description
                "The interval between periodic transmissions of
                configuration messages by designated ports";
        }

        leaf max_age {
            type uint8 {
                range "6..40" {
                    error-message "Invalid Maximum Age Time value.";
                }
            }
            units seconds;
            default 20;
            description
                "The maximum age of the information transmitted by the
                bridge when it is the root bridge";
        }

        leaf priority {
            type uint16 {
                range "0..61440" {
                    error-message "Invalid Bridge Priority value.";
                }
            }
            sonic-ext:custom-validation ValidateBridgePriority;
            default 32768;
            description
                "The manageable component of the Bridge Identifier";
        }
    }

Add this , make add as below on this block
  uses vlanModeAttr;

Reply:
Please review, I have updated